server/container_create_linux.go

```diff
@@ -534,6 +537,15 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID, contai
 		}
 		specgen.SetLinuxCgroupsPath(cgPath)
 
+		if t, ok := kubeAnnotations[podTerminationGracePeriodLabel]; ok {
+			// currently only supported by systemd, see
+			// https://github.com/opencontainers/runc/pull/2224
+			if useSystemd {
+				specgen.AddAnnotation("org.systemd.property.TimeoutStopUSec",
+					"uint64 "+t+"000000") // sec to usec
```

Comment:
I think we also want to protect ourselves from pods setting arbitrarily high timeouts delaying shut down so we could limit this to something like 5 minutes.

Reply:
Not sure it's a good idea to do that kind of validation in here. Such kind of validation belongs to either upper level, i.e. kubernetes, which validates input from user, or lower levels, i.e. systemd, which might find some values inapplicable, violating its settings, or otherwise bad.

In ths case, cri-o is just a messenger, and a messenger should not be overly smart.

Reply:
I agree with you and I would prefer it to be at a higher level, too. It could be an admission hook or something like a Limit Range. However, either of those options may take time to deliver, so this suggestion is for short term where we can ship this (to help with issues we have at shutdown) without a bug opened for updates getting blocked by rogue pods.

Reply:
Makes sense (with something like `// TODO remove this once kubernetes PR #xxx is merged`).

In case of unreasonably high value (`val > MAX`), are you suggesting to
- return an error,
- skip the setting, or
- silently set it to `MAX`?

Reply:
We can log a warning and set it to MAX if a value is set > MAX. Thanks!